wm/position.go

```diff
@@ -0,0 +1,20 @@
+package wm
+
+import (
+	"fyne.io/fynedesk"
+	"fyne.io/fynedesk/theme"
+)
+
+// PositionForNewWindow returns the suggested position for a new window of the given geometry.
+// The screen list hints at available space, but normally list.Active() is the best.
+func PositionForNewWindow(x, y int, w, h uint, screens fynedesk.ScreenList) (int, int, uint, uint) {
+	if x != 0 && y != 0 {
+		return x, y, w, h
+	}
+
+	target := screens.Active()
```

Comment:
Here let's go ahead and do something like this:
winType = windowTypeGet(c.wm.X(), win)
transient = false
transientFor = x11.WindowTransientForGet(c.wm.X(), win)
if transientFor > 0 && transientFor != win {
     transient = true
}
if winType == windowTypeNormal && !transient {
     * We know this is a top level window and place top left *
} else if winType == windowTypeDialog && transient {
     * center win on transientFor *
} else {
     * try and use provided x, y geometry (and fix if broken) because this could be a dnd window, menu, utility that needs special placement, etc...
}

Reply:
Hmm. We don't expose transientFor properties through the API currently... or type for that matter.
Could that be added in a later PR when we figure how to write that in a non-X specific way?